docs/Caching.md

```diff
@@ -2,16 +2,36 @@
+The portal is configured to use Ehcache for backend caching. Ehcache supports a hybrid (disk + heap), disk-only, and heap-only mode; this, along with additional specifications such as cache size and location, are set inside `portal.properties`(more information [here](portal.properties-Reference.md#ehcache-settings)). 
  
 ## Creating additional caches
-The default configuration initializes two separate caches. However, you may wish to introduce new caches with different policies (e.g expiration policy) for different datatypes. To create additional caches (e.g creating a cache specifically for clinical data), a new cache must be added to the Ehcache xml configuration file. The `@Cacheable` annotation must also be added (or adjusted) to function declarations to indicate which functions are to be cached. Those might look like this example:
+Cache initialization is handled inside the [CustomEhCachingProvider](../persistence/persistence-api/src/main/java/org/cbioportal/persistence/util/CustomEhCachingProvider.java). The default configuration initializes two seperate caches; however, you may wish to introduce new caches for different datatypes. To create additional caches (e.g creating a cache specifically for clinical data), a new cache must be added to the CustomEhCachingProvider. 
+
+Within the `CustomEhCachingProvider`, initialize a new `ResourcePoolsBuilder` for the new cache and set the resources accordingly. 
+```
+ResourcePoolsBuilder clinicalDataCacheResourcePoolsBuilder = ResourcePoolsBuilder.newResourcePoolsBuilder();
+clinicalDataCacheResourcePoolsBuilder.heap(clinicalDataCacheHeapSize, MemoryUnit.MB);
+clinicalDataCacheResourcePoolsBuilder.disk(clinicalDataCacheDiskSize, MemoryUnit.MB);
+```
+After initialzing the `ResourcePoolsBuilder`, create a CacheConfiguration for the new cache using the new `ResourcePoolsBuilder` just created.
+```
+CacheConfiguration<Object, Object> clinicalDataCacheConfiguration = xmlConfiguration.newCacheConfigurationBuilderFromTemplate("RepositoryCacheTemplate", 
+Object.class, Object.class, clinicalDataCacheResourcePoolsBuilder)
+.withSizeOfMaxObjectGraph(Long.MAX_VALUE)
+.withSizeOfMaxObjectSize(Long.MAX_VALUE, MemoryUnit.B)
+.build();
+```
+Finally, add the new `CacheConfiguration` to the map of managed caches with a name for the cache. 
+```
+caches.put("ClinicalDataCache", ClinicalDataCacheConfiguration);
+```
+The `@Cacheable` annotation must also be added (or adjusted) to function declarations to indicate which functions are to be cached. Those might look like this example:
 ``` 
 @Cacheable(cacheNames = "ClinicalDataCache", condition = "@cacheEnabledConfig.getEnabled()")
 public String getDataFromClinicalDataRepository(String param) {}
 ```
 Additionally, new properties for setting cache sizes should be added to `portal.properties`. Alternatively, values may also be hardcoded directly into the Ehcache xml configuration file. 
```

Comment:
Maybe here we should say:

Additionally, new properties for setting cache sizes should be added to `portal.properties` -->

Additionally, new properties for setting cache sizes should be added to `portal.properties` and loaded into [CustomEhCachingProvider](../persistence/persistence-api/src/main/java/org/cbioportal/persistence/util/CustomEhCachingProvider.java).

Or just remove this line?  And the one that follows.